src/Types/ContextFactory.php

```diff
@@ -70,6 +70,26 @@ public function createForNamespace($namespace, $fileContents)
                 case T_NAMESPACE:
                     $currentNamespace = $this->parseNamespace($tokens);
                     break;
+                case T_CLASS:
+                    // Fast-forward the iterator through the class so that any
+                    // T_USE tokens found within are skipped - these are not
+                    // valid namespace use statements so should be ignored.
+                    $braceLevel = 0;
+                    $firstBraceFound = false;
+                    while ($tokens->valid() && ($braceLevel > 0 || !$firstBraceFound)) {
+                        if ($tokens->current() === '{') {
```

Comment:
@nikic it may not be valid PHP and not be able to execute, but it seems that if you perform a `token_get_all()` on this, it will still give you valid tokens (even if it won't execute):

```php
namespace A {
  class Foo {
    function a() {
      echo 'A\Foo';
    }
  }
}

namespace B {
  class Foo extends Bar{
    function a() {
      echo 'B\Foo';
    }
  }
  use A\Foo as Bar;
}
```

In this edge case where tokenizer will happily tokenize this, the depth-checking method I've implemented (and improved in a soon-to-be-opened-PR) would be safer.